packages/dd-trace/src/proxy.js

```diff
@@ -121,6 +108,27 @@ class Tracer extends NoopProxy {
     return this
   }
 
+  _enableOrDisableTracing (config) { // TODO: add test
+    if (config.tracing) {
+      // dirty require for now so zero appsec code is executed unless explicitly enabled
+      if (config.appsec.enabled) {
+        require('./appsec').enable(config)
+      }
+      if (!this._tracingInitialized) {
+        this._tracer = new DatadogTracer(config)
+        this.appsec = new AppsecSdk(this._tracer, config)
+        this._tracingInitialized = true
+      }
+      if (config.iast.enabled) {
+        require('./appsec/iast').enable(config, this._tracer) // disable
+      }
+      this._pluginManager.configure(config)
+    } else {
```

Comment:
This is missing any code needed to disable other things. While it's possible that updating the config object alone would work, it's best to be safe and explicitly reconfigure everything.

Reply:
Basically, same as above. The `configure` calls on the tracer and the plugin manager should be happening in this function.